main.go

```diff
@@ -12,18 +12,28 @@ func main() {
 	interrupt := make(chan os.Signal, 1)
 	signal.Notify(interrupt, os.Interrupt)
 
+	// it is better that each entity will do its own setup. Less parameters when you call entities.NewXXXXX
+	// it will also be clearer which values are needed for each entity.
+	// will be great if you have a different setup for each entity (.env and config.json can be common).
 	app := setup()
 
 	shutdownChan := make(chan struct{})
+	// swapIntentOperationSubmitChan used by user to submit swapIntentOperations to bundler
 	swapIntentOperationSubmitChan := make(chan *entities.SwapIntentOperation)
+	// swapIntentOperationBroadcastChan used by bundler to announce swapIntentOperations to solver
 	swapIntentOperationBroadcastChan := make(chan *entities.SwapIntentOperation)
+	// solverOperationSubmitChan used by solver to submit solverOperation back to bundler
 	solverOperationSubmitChan := make(chan *Atlas.SolverOperation)
+	// governanceSignatureChan - ??????
 	governanceSignatureChan := make(chan []byte)
 
+	// Q: what is the governance entity? It is not clear why governance is needed and when?
+	// Q: is this an online component (service) that the dapp has to provide?
 	// Launch the governance goroutine
 	governance := entities.NewGovernance(app.PrivateKeys["governance"], governanceSignatureChan, shutdownChan)
 
-	// Launch the backend goroutine
+	// Launch the backend (bundler) goroutine
+	// Q: the backend is sending to governance and governance is sending back to backend - how can they do it on the same channel? looks wrong
 	entities.NewBackend(app.PrivateKeys["bundler"], app.EthClient, app.Atlas, app.DAppController, app.TxBuilder, governance.GetGovernanceAddress(),
 		swapIntentOperationSubmitChan, swapIntentOperationBroadcastChan, solverOperationSubmitChan, governanceSignatureChan, shutdownChan)
```

Comment:
In this example, the backend is treating intents synchronously, so using one channel for request/response works.

Reply:
I'm not sure I agree here. Looks like a bug to me.

Reply:
The script is running fine for me. Did you run into issues when running it?

Reply:
No, it was running for me ok, but this is probably due to luck/no load.
if you have A and B both reading and writing to a channel. 
There is nothing that prevents A to read what A just wrote to the channel. Same for B.